ow_faults/src/FaultInjector.cpp

```diff
@@ -93,24 +104,62 @@ void FaultInjector::antennaePanFaultCb(const std_msgs::Float64& msg){
   publishAntennaeFaults(msg, 
                         m_faults.ant_pan_encoder_failure, 
                         m_faults.ant_pan_effort_failure, 
-                        m_faultPanValue, m_fault_ant_pan_pub );
+                        m_faultPanValue, m_fault_ant_pan_remapped_pub );
 }
 
 void FaultInjector::antennaeTiltFaultCb(const std_msgs::Float64& msg){
   publishAntennaeFaults(msg, 
                         m_faults.ant_tilt_encoder_failure, 
                         m_faults.ant_tilt_effort_failure, 
-                        m_faultTiltValue, m_fault_ant_tilt_pub );
+                        m_faultTiltValue, m_fault_ant_tilt_remapped_pub );
 }
 
-void FaultInjector::setPowerTemperatureFaultValue(bool getTempBool){
-  float thermal_val;
-  if (isnan(powerTemperatureOverloadValue)) {
-    thermal_val =  50.0 + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(90.0-50.0)));
-    powerTemperatureOverloadValue = thermal_val;
-  } else if (!getTempBool) {
-    powerTemperatureOverloadValue = NAN;
+float FaultInjector::getRandomFloatFromRange( float min_val, float max_val){
+  return min_val + (max_val - min_val) * (rand() / static_cast<float>(RAND_MAX));
+}
+
+void FaultInjector::publishPowerSystemFault(){
+  //power
+  ow_faults::PowerFaults power_faults_msg;
+  ComponentFaults hardwareFault =  ComponentFaults::Hardware;
+  //system
+  std::bitset<10> systemFaultsBitmask{};
+  ow_faults::SystemFaults system_faults_msg;
+
+  systemFaultsBitmask |= systemFaultsBitset;
+  //update if fault
+  if (m_temperatureFault || m_socFault) {
+    //system
+    systemFaultsBitmask |= isPowerSystemFault;
+    systemFaultsBitset = systemFaultsBitmask;
+    //power
+    setComponentFaultsMessage(power_faults_msg, hardwareFault);
   }
+  //publish
+  setBitsetFaultsMessage(system_faults_msg, systemFaultsBitmask);
+  m_system_fault_jpl_msg_pub.publish(system_faults_msg);
+  m_power_fault_jpl_msg_pub.publish(power_faults_msg);
+
+}
+
+void FaultInjector::powerTemperatureListener(const std_msgs::Float64& msg)
+{
+  m_temperatureFault = ( msg.data > THERMAL_MAX);
+  publishPowerSystemFault();
+
+}
+
+void FaultInjector::powerSOCListener(const std_msgs::Float64& msg)
+{
+  float newSOC = msg.data;
+  if (isnan(m_originalSOC)){
```

Comment:
to make use of the isnan function you need to initialize m_originalSOC - which should be m_last_SOC -  to **std::numeric_limits<float>::quiet_NaN()**